tests/publiccloud/migration.pm

```diff
@@ -0,0 +1,112 @@
+# SUSE's openQA tests
+#
+# Copyright © 2018-2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Test distribution migration system in public cloud SUSE images
+#
+# Maintainer: Jesus Bermudez <jesus.bv@suse.com>
+
+use Mojo::Base 'publiccloud::basetest';
+use testapi;
+use Mojo::File 'path';
+use Mojo::JSON;
+
+our $custom                 = get_required_var('CUSTOM');
+our $distro_name            = get_required_var('DISTRO_NAME');
+our $custom_distro_name     = get_required_var('CUSTOM_DISTRO_NAME');
+our $version                = get_required_var('CUSTOM_VERSION');
+our $arch                   = get_required_var('ARCH');
+our $do_not_migrate         = get_required_var('DO_NOT_MIGRATE');
+our $default_target_version = get_required_var('DEFAULT_TARGET_VERSION');
+our $default_origin_version = get_required_var('DEFAULT_ORIGIN_VERSION');
+
+sub install_package {
+    my ($self, %args) = @_;
+    my $instance = $args{instance};
+
+    record_info('INFO', 'Installing distribution migration packages');
+    $instance->run_ssh_command(
+        cmd      => 'sudo zypper in -y SLES15-Migration',
+        no_quote => 0,
+        timeout  => 30
+    );
+    $instance->run_ssh_command(
+        cmd      => 'sudo zypper in -y suse-migration-sle15-activation',
+        no_quote => 0,
+        timeout  => 30
+    );
+}
+
+sub target_version {
+    my ($self, %args) = @_;
+    my $instance       = $args{instance};
+    my $custom_product = $distro_name . '/' . $version . '/' . $arch;
+    record_info('INFO', 'Setting custom product to migrate');
+    record_info('INFO', $custom_product);
+    # tee works like > # tee -a works like >>
```

Comment:
not sure that we really need this comment

Reply:
But I like it 😄 I was not using `tee` before I started to write openQA tests.